src/state_machine.zig

```diff
@@ -22,10 +22,11 @@ const CreateTransfersResult = tb.CreateTransfersResult;
 const CreateAccountResult = tb.CreateAccountResult;
 const CreateTransferResult = tb.CreateTransferResult;
 
-pub fn StateMachineType(comptime Storage: type, comptime constants_: struct {
-    message_body_size_max: usize,
-    lsm_batch_multiple: usize,
-}) type {
+pub fn StateMachineType(comptime Storage: type, comptime constants_: type) type {
+    assert(constants_.message_body_size_max > 0);
+    assert(constants_.lsm_batch_multiple > 0);
+    assert(constants_.vsr_operations_reserved > 0);
```

Comment:
Note sure how I feel about this, we seem to lose some type-safety here. Perhaps we can

```zig
pub fn StateMachineType(comptime Storage: type, comptime constants_: struct {
    message_body_size_max: usize = @import("constants").message_body_size_max,
}) type {
```

?